code/game/objects/items/candle.dm

```diff
@@ -69,6 +70,32 @@
 		processing_objects.Add(src)
 		update_icon()
 
+/obj/item/candle/proc/flicker(var/amount = rand(5, 15))
+	if(flickering)
+		return
+	flickering = 1
+	spawn(0)
+		if(lit)
+			for(var/i = 0; i < amount; i++)
+				lit = 0
+				update_icon()
+				sleep(rand(3,9))
+				lit = 1
+				update_icon()
+			flickering = 0
+
+/obj/item/candle/spook(mob/dead/observer/ghost)
+	if(..(ghost, TRUE))
+		flicker()
+
+/obj/item/candle/attack_ghost(mob/user)
+	if(!can_spook())
+		return
+	src.add_hiddenprint(user)
+	src.flicker(1)
```

Comment:
no need for sec

Reply:
you sure? I copypasted this (and removed the wall light-specific parts) from lighting.dm, that's how it's implemented there

Reply:
certain, src just refers to self